src/Collection/CollectionTrait.php

```diff
@@ -658,4 +658,23 @@ public function _unwrap()
     {
         return $this->unwrap();
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return \Cake\Collection
+     */
+    public function transpose()
+    {
+        $arrayValue = $this->toArray();
```

Comment:
I think this should be `toList` instead of `toArray`. The reason is that using keys main considerably alter the result of this method to a ways that would be almost impossible to prevent by the user. Since we are dropping the keys in this method anyway, I think it is OK to use that.